lib/GADS/Schema/ResultSet/User.pm

```diff
@@ -255,7 +240,7 @@ sub upload
     # Won't get this far if we have any errors in the previous statement
     $guard->commit;
 
-    $self->_send_welcome_email(%$_, request_base => $options{request_base})
+    Schema::Result::User::send_welcome_email(%$_, request_base => $options{request_base})
         foreach @welcome_emails;
```

Comment:
Sorry to be a pain... looking at this afresh, I think that `@welcome_emails` should be changed to be an array of `Schema::Result::User` objects (i.e. change above to be `push @welcome_emails, $u`) and then the above could be written as:
```
$_->send_welcome_email(code => $_->resetpw, request_base => $options{request_base})
    foreach @welcome_emails;
```
It's basically the same, I just think it's generally better to call the method `send_welcome_email` method natively.

Or something like that - it would need testing.

Reply:
Ironically, I was thinking the same a moment ago, then I thought I'd see if you thought the same (you did).